client/lib/wp/sync-handler/index.js

```diff
@@ -51,6 +91,7 @@ export class SyncHandler {
 			// create a copy of the request params
 			const reqParams = Object.assign( {}, params );
 			const { path } = reqParams;
+			let localResponseBody = null;
```

Comment:
The most annoying part about promises is that we can only resolve or reject with a single argument. I wonder if it'd be worthwhile to pass through a larger object, so we can avoid these closures
```javascript
{
    localResponseBody: null,
    response: null
}
```